### PR TITLE
Fix computation of available memory on gpu (avoid truncation and conversions)

### PR DESCRIPTION
**Pull request by abouteiller**
Fix computation of available memory on gpu (avoid truncation and conversions) and add the amount of reserved mem to the device caps output

### CODE REVIEW

**Review by devreal — APPROVED** [review]
LGTM

File: parsec/mca/device/cuda/device_cuda_module.c [hunk 1]
```diff
@@ -575,23 +575,15 @@ parsec_cuda_module_init( int dev_id, parsec_device_module_t** module )
     }
 
     if( show_caps ) {
-        parsec_inform("GPU Device %-8s: %s [capability %d.%d] %s\n"
```
Comment (bosilca) [inline parsec/mca/device/cuda/device_cuda_module.c:578]: not sure I understand the need to change the device info.

File: parsec/mca/device/device_gpu.c [hunk 2]
```diff
@@ -597,9 +597,9 @@ parsec_device_memory_reserve( parsec_device_gpu_module_t* gpu_device,
     int rc;
     (void)eltsize;
 
-    size_t how_much_we_allocate;
+    size_t alloc_size;
     size_t total_mem, initial_free_mem;
-    uint32_t mem_elem_per_gpu = 0;
```
Comment (bosilca) [inline parsec/mca/device/device_gpu.c:602]: Apart renaming and reordering the code, what is the real change here ? The shift for the local variable from uint32_t to size_t ? Why is this needed ?

Comment (abouteiller) [reply parsec/mca/device/device_gpu.c:602]: the test for how_much_we_allocate > initial_free_mem was done twice, in similar but not exactly the same way. The second time, around line 699, the use of int32_t would cause the result to be random (truncated) on some large memory systems.

File: parsec/mca/device/cuda/device_cuda_module.c (hunk 1 above)
Comment (abouteiller) [reply parsec/mca/device/cuda/device_cuda_module.c:578]: The main goal was to add the amount of actually allocated memory (and the total capacity), the rest of the reformatting is just cosmetic to reduce the number of output lines, the old format was pretty annoying on Frontier with 8GPUs per node.

**Review by bosilca — APPROVED** [review]
